checkpoint/orbax/checkpoint/_src/serialization/replica_slices.py

```diff
@@ -141,26 +142,26 @@ def _hashable_slices(slices: Index) -> tuple[HashableSlice, ...]:
   return tuple([(s.start, s.stop, s.step) for s in slices])
 
 
-def _create_replica_counts_builder(
-    arr: jax.Array,
-) -> Callable[[jax.Shard], int]:
-  """Produces a mapping from addressable shards to global replication count."""
-
+@jax.util.cache(max_size=4096, trace_context_in_key=False)
```

Comment:
Don't use jax cache, that's private 

Use functools.lru_cache

Reply:
Done. How do I tell which APIs are private?

Reply:
It's documented here: https://jax.readthedocs.io/en/latest/jax.html